src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskAssignByIdControllerTest.java

```diff
@@ -35,155 +36,191 @@ public void setUp() {
 
     @Test
     public void should_return_a_404_if_task_does_not_exist() {
-        String nonExistentTaskId = "00000000-0000-0000-0000-000000000000";
-
-        common.setupOrganisationalRoleAssignment(authenticationHeaders);
-
-        Response result = restApiActions.post(
-            ENDPOINT_BEING_TESTED,
-            nonExistentTaskId,
-            new AssignTaskRequest(assigneeId),
-            authenticationHeaders
-        );
-
-        result.then().assertThat()
-            .statusCode(HttpStatus.NOT_FOUND.value())
-            .and()
-            .contentType(APPLICATION_JSON_VALUE)
-            .body("timestamp", lessThanOrEqualTo(ZonedDateTime.now().plusSeconds(60)
-                .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
-            .body("error", equalTo(HttpStatus.NOT_FOUND.getReasonPhrase()))
-            .body("status", equalTo(HttpStatus.NOT_FOUND.value()))
-            .body("message", equalTo(String.format(
-                LOG_MSG_THERE_WAS_A_PROBLEM_FETCHING_THE_VARIABLES_FOR_TASK,
-                nonExistentTaskId
-            )));
+        boolean isFeatureEnabled = launchDarklyClient.getKey(
+            FeatureFlag.RELEASE_2_ENDPOINTS_FEATURE.getKey(),
+            authenticationHeaders.getValue(AUTHORIZATION));
+
+        if (!isFeatureEnabled) {
+            String nonExistentTaskId = "00000000-0000-0000-0000-000000000000";
```

Comment:
I believe this test should be applicable for both r1 and r2 if a taks does not exist and we try to assign it a 404 should be thrown